examples/github/main.tf

```diff
@@ -66,26 +73,33 @@ data "kubectl_file_documents" "install" {
   content = data.flux_install.main.content
 }
 
-resource "kubectl_manifest" "install" {
-  for_each   = { for v in data.kubectl_file_documents.install.documents : sha1(v) => v }
-  depends_on = [kubernetes_namespace.flux_system]
-
-  yaml_body = each.value
-}
-
 data "kubectl_file_documents" "sync" {
   content = data.flux_sync.main.content
 }
 
-resource "kubectl_manifest" "sync" {
-  for_each   = { for v in data.kubectl_file_documents.sync.documents : sha1(v) => v }
-  depends_on = [kubectl_manifest.install, kubernetes_namespace.flux_system]
+locals {
+  install = [ for v in data.kubectl_file_documents.install.documents : {
+      data: yamldecode(v)
+      content: v
+    }
+  ]
+  sync = [ for v in data.kubectl_file_documents.sync.documents : {
+      data: yamldecode(v)
+      content: v
+    }
+  ]
+}
 
+resource "kubectl_manifest" "install" {
+  for_each   = { for v in local.install : lower(join("/", compact([v.data.apiVersion, v.data.kind, lookup(v.data.metadata, "namespace", ""), v.data.metadata.name]))) => v.content }
```

Comment:
Nice way of handling it! Maybe it's easier to read if it's a parameter of the local and not part of the resource?

Reply:
I am trying to avoid too many loops for now.